src/pkgcheck/checks/codingstyle.py

```diff
@@ -79,6 +79,29 @@ def feed(self, pkg):
                     yield DeprecatedEapiCommand(name, line=call, lineno=lineno+1, eapi=pkg.eapi, pkg=pkg)
 
 
+class EendMissingArg(results.LineResult, results.Style):
+    """Eend is called without an argument"""
+
+    @property
+    def desc(self):
+        return f'Call to eend is missing an argument, on line {self.lineno}'
+
+
+class EendMissingArgCheck(Check):
+    """Scan an ebuild for calls to eend without an argument"""
+
+    _source = sources.EbuildParseRepoSource
+    known_results = frozenset([EendMissingArg])
+
+    def feed(self, pkg):
+        for func_node, _ in bash.func_query.captures(pkg.tree.root_node):
+            for node, _ in bash.cmd_query.captures(func_node):
+                call = pkg.node_str(node)
+                if call == "eend":
```

Comment:
I'm sure with the right s-expression, this could be reduced to one query, iterating over the result and reporting each hit.